### ISSUE TITLE
client.addLibrary fails with 'Invalid CEN Header (Bad Signature)'

### ISSUE THREAD

**Issue by nawgszy** [webgme/webgme-engine#126]
I've just updated to 2.21.0. As discussed in #63, I wished to add a seed as a library. You added this functionality, and I am just now updating my code to use it. I can check if the version I was using previously worked correctly, if it seems important.

My createProject code in my projectStore class on the front-end looks something like this (I wrapped most of the client functionality with promises, so bear with me that it wouldn't run quite like this):

```
createProject(projectName, projectType) {
  client.seedProject({
    type: type,
    projectName: projectName,
    seedBranch: 'master',
    seedName: seedName,
    ownerId: ownerId,
  })
  .then((result) => this._loadProject(result.projectId)) // we're good until here
  .then(() => fetch('/api/seeds/:seedName'))
  .then((body) => {
    // we correctly get the hash for the seed here
    const seedHash = body.blobHash;

    return client.addLibrary(projectType, seedHash);
  })
  .catch((err) => console.error(err)) // this is where I see the Invalid CEN Header
```

This all worked flawlessly when I created my own router with its own blobClient and added the seeds to them in that router on server startup. I am not sure what is different, but now it fails:

The file `webgme-engine/src/client/gmeServerRequests.js` has a member `addLibrary`. This finds its way onto the backend where the file `webgme-engine/src/server/worker/workerrequests.js` has a member `addLibrary`. Inside this `addLibrary` function, there is a call to `_getProjectJsonFromBlobHash`, which gets a `Buffer` from `blobClient.getObject`, passes it to `_extractProjectJsonAndAddAssets`, which fails at the line:

```
var zip = new AdmZip(filenameOrBuffer);
```

When I take that seedHash I get from `/api/seeds/:seedName` I can correctly download the `:seedName.webgmex` file, so I can't really understand what the issue is here. Any help would be greatly appreciated.

**Comment by pmeijer** [comment]
This is fixed in #127. Thanks for the detailed explanation - make it much easier to find the bug.
Turns out I made a mistake by decoding the zip file before uploading it. There were tests checking that a blob was uploaded and a test checking that addLibrary worked with a blob. However these weren't tested together.
Do you use webgme or webgme-engine directly? If the latter we can release the engine today.

**Comment by nawgszy** [comment]
I am using `webgme-engine` directly. A release would be greatly appreciated. Thanks!
